pkg/gitrepo/repo.go

```diff
@@ -119,6 +121,12 @@ func (r *Repo) EnsureUpToDate(ctx context.Context) error {
 	err = repo.Fetch(fetchOpts)
 	if errors.Is(err, git.NoErrAlreadyUpToDate) {
 		// Fall through.
+	} else if errors.Is(err, transport.ErrRepositoryNotFound) {
+		// This could happen if the repository does not exist, but you already have the folder on the filesystem.
+		// In that case Fetch will be the first to realise that repo does not exist since Clone only performs an Open.
+		// Also, Clone creates the folder on the filesystem even if it fails so you end simulate the same situation when
+		// you call EnsureUpToDate more that once on the same non-existent repo.
+		return microerror.Maskf(repositoryNotFoundError, "%#q", r.url)
```

Comment:
I wonder if we even shouldn't translate this error to something like repository in broken state, because subsequent call won't work as well.

Reply:
I think this matches the actual reason why it failed - the repository does not exist. If the repo was there, then it would've recovered by fetching the latest state.

Reply:
So you say subsequent call to `EnsureUpToDate` can succeed?

Reply:
If you had a successful Clone in the past and the repo magically reappears, I assume it will (haven't tried it out). 

Also, you don’t have a guarantee that the repo is borked here. It could be just that you cloned it a long time ago and eventually someone deleted the remote repo. So Fetch needs to handle this error anyway.

It’s probably nice to add something explicit telling the user that Clone failed and you could have an invalid directory left behind. But that includes distinguishing if Clone failed during or after initialising the repo. So I'm not sure it's really worth the effort.